view/frontend/templates/js/replacejs.phtml

```diff
@@ -2279,6 +2284,30 @@ if (!$block->isSaveCartInSections()) { ?>
         ////////////////////////////////////////////////////
         <?php echo $block->getAdditionalJavascript(); ?>
     });
+
+    if (trim(location.pathname, '/') === 'checkout/cart') {
+        require([
+        'jquery',
+        'Magento_Customer/js/customer-data',
+        'Magento_Checkout/js/model/quote',
+        'domReady!'
+        ], function ($, customerData, quote) {
+            quote.totals.subscribe(function (newValue) {
+                if (expectCartRendering) {
+                    expectCartRendering = false;
+                    return;
+                }
+
+                // If we don't expect cart rendering possible it caused by any third party plugin (for example, 3p coupon is applied)
```

Comment:
Can we bubble up this comment to the top of this function and explain a bit more in detail what the whole thing is doing, something like "If quote total was changed through a method we didn't create, we should reload the bolt cart since totals are probably now out of sync"

Reply:
Done